### PR TITLE
fix(bitbucket-cloud): Preserve Bitbucket Cloud reviewers when updating PRs

### LINKED ISSUE #5772: [Bug] bb-cloud: changing reviewers leads to changed body
**What Renovate type are you using?**

self-hosted on bitbucket cloud

**Describe the bug**

when changes on the PR are made (i.e.: reviewer modified) the body is being changed by bitbucket while saving the PR.
In front of each `\n` a `\r` is inserted on PR editing. (strange thing to do, imho, but we are talking about an Atlassian tool afterall)

After manual edit of the PR (added one reviewer):

       "oldPrBody": "This PR contains the following updates:\r\n\r\n| Package | Update | Change |\r\n| --- | --- | --- |\r\n| [org.mockito:mockito-core](https://github.com/mockito/mockito) \\([source](https://github.com/mockito/mockito)\\) | major | `2.28.2` -> `3.0.0` |\r\n\r\n---\r\n\r\n### Renovate configuration\r\n\r\n:date: **Schedule**: At any time \\(no schedule defined\\).\r\n\r\n:vertical_traffic_light: **Automerge**: Disabled by config. Please merge this manually once you are satisfied.\r\n\r\n:recycle: **Rebasing**: Whenever PR becomes conflicted.\r\n\r\n:no_bell: **Ignore**: Close this PR and you won't be reminded about this update again.\r\n\r\n---\r\n\r\nThis PR has been generated by [Renovate Bot](https://renovatebot.com).",

what renovate generates:

       "newPrBody": "This PR contains the following updates:\n\n| Package | Update | Change |\n|---|---|---|\n| [org.mockito:mockito-core](https://github.com/mockito/mockito) ([source](https://github.com/mockito/mockito)) | major | `2.28.2` -> `3.0.0` |\n\n---\n\n### Renovate configuration\n\n:date: **Schedule**: At any time (no schedule defined).\n\n:vertical_traffic_light: **Automerge**: Disabled by config. Please merge this manually once you are satisfied.\n\n:recycle: **Rebasing**: Whenever PR becomes conflicted.\n\n:no_bell: **Ignore**: Close this PR and you won't be reminded about this update again.\n\n---\n\nThis PR has been generated by [Renovate Bot](https://renovatebot.com)."

This leads to #4515 happening

**Did you see anything helpful in debug logs?**

see quoted bodys above

**To Reproduce**

* create bitbucket-cloud repo
* commit `package.json` with to be updated content
* wait for renovate to do its job
* add a reviewer to the PR and hit save
* wait for renovate to revisit this PR

**Additional context**

### PR DESCRIPTION
**Pull request by daekene**
This PR fixes an issue where Renovate will remove the reviewers associated with a Bitbucket Cloud PR when it interacts with it after creation. The issue here is that unless the reviewers are sent alongside the update request, they will be cleared away. The fix is to fetch the current list of reviewers and send them alongside any changes being made to the PR.

Closes #5772

### CODE REVIEW

File: lib/platform/bitbucket/index.ts [hunk 1]
```diff
@@ -332,6 +332,7 @@ interface PrResponse {
       name: string;
     };
   };
+  reviewers: Array<any>;
```
Comment (JamieMagee) [inline lib/platform/bitbucket/index.ts:335]: Can we be more specific here? Looks like `reviewers: {username: string}[]` might be more appropriate.

**Comment by JamieMagee** [comment]
Is this something that also needs to be fixed in Bitbucket server?

**Comment by viceice** [comment]
@JamieMagee Should already be fixed in bb server.

**Review by viceice — COMMENTED** [review]
Please fix the types, otherwise LGTM

**Review by rarkins — CHANGES_REQUESTED** [review]
Please ensure that the test case does test the new functionality and sends back the list of reviewers in the format required.

File: lib/platform/bitbucket/index.ts (hunk 1 above)
Comment (daekene) [reply lib/platform/bitbucket/index.ts:335]: The type returned here is a bit more complicated than that. It can return different fields, some that are populated and some that are not. I felt that a simple solution would be to put `any` here since we don't really care what the values are, we just immediately return them to Bitbucket in the very next call.

Here's are links to the relevant API docs: 
- Pull request response: https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Bworkspace%7D/%7Brepo_slug%7D/pullrequests/%7Bpull_request_id%7D
- User/account object: https://developer.atlassian.com/bitbucket/api/2/reference/resource/user

```
reviewers0..∞

The list of users that were added as reviewers on this pull request when it was created. For performance reasons, the API only includes this list on a pull request's self URL.
all of
object

Base type for most resource objects. It defines the common type element that identifies an object's type. It also identifies the element as Swagger's discriminator.
typestring

An account object.
...
```

File: lib/platform/bitbucket/index.spec.ts [hunk 2]
```diff
@@ -844,10 +844,24 @@ describe('platform/bitbucket', () => {
   describe('updatePr()', () => {
     it('puts PR', async () => {
       const scope = await initRepoMock();
-      scope.put('/2.0/repositories/some/repo/pullrequests/5').reply(200);
+      scope
+        .get('/2.0/repositories/some/repo/pullrequests/5')
+        .reply(200, { reviewers: [] })
+        .put('/2.0/repositories/some/repo/pullrequests/5')
```
[the remark this answers is not in the document]
Comment (rarkins) [reply lib/platform/bitbucket/index.spec.ts:850]: We need a test case where Bitbucket returns a non-empty list of reviewers and then we `put` the result.

[the remark this answers is not in the document]
Comment (daekene) [reply lib/platform/bitbucket/index.spec.ts:784]: Sorry, I was busy for the past week and didn't have time to follow this up. I've added a reviewer to the GET payload, which is then PUT back and is visible in the snapshot file. I hope that's an acceptable test level

File: lib/platform/bitbucket/index.ts (hunk 1 above)
Comment (viceice) [reply lib/platform/bitbucket/index.ts:284]: If we don't care, we should use `unknown` instead of `any`
